Source/Core/Core/State.cpp

```diff
@@ -297,58 +317,24 @@ static std::map<double, int> GetSavedStates()
   return m;
 }
 
-struct CompressAndDumpState_args
-{
-  std::vector<u8>* buffer_vector = nullptr;
-  std::mutex* buffer_mutex = nullptr;
-  std::string filename;
-  bool wait = false;
-};
-
-static void CompressAndDumpState(CompressAndDumpState_args save_args)
+static void CompressAndDumpState(CompressAndDumpState_args& save_args)
 {
-  std::lock_guard lk(*save_args.buffer_mutex);
-
-  // ScopeGuard is used here to ensure that g_compressAndDumpStateSyncEvent.Set()
-  // will be called and that it will happen after the IOFile is closed.
-  // Both ScopeGuard's and IOFile's finalization occur at respective object destruction time.
-  // As Local (stack) objects are destructed in the reverse order of construction and "ScopeGuard
-  // on_exit"
-  // is created before the "IOFile f", it is guaranteed that the file will be finalized before
-  // the ScopeGuard's finalization (i.e. "g_compressAndDumpStateSyncEvent.Set()" call).
-  Common::ScopeGuard on_exit([]() { g_compressAndDumpStateSyncEvent.Set(); });
-  // If it is not required to wait, we call finalizer early (and it won't be called again at
-  // destruction).
-  if (!save_args.wait)
-    on_exit.Exit();
-
-  const u8* const buffer_data = &(*(save_args.buffer_vector))[0];
-  const size_t buffer_size = (save_args.buffer_vector)->size();
-  std::string& filename = save_args.filename;
-
-  // For easy debugging
-  Common::SetCurrentThreadName("SaveState thread");
-
-  // Moving to last overwritten save-state
-  if (File::Exists(filename))
+  const u8* const buffer_data = save_args.buffer_vector.data();
+  const size_t buffer_size = save_args.buffer_vector.size();
+  const std::string& filename = save_args.filename;
+
+  // Find free temporary filename.
+  // TODO: The file exists check and the actual opening of the file should be atomic, we don't have
```

Comment:
Will this TODO be implemented before the review is merged or fixed up later?

Reply:
I'm not planning to implement this in this PR -- it would involve writing an entire separate IOFile-like class for Windows.